src/Parallel/GlobalCache.hpp

```diff
@@ -46,29 +47,7 @@ template <class GlobalCacheTag, class Metavariables>
 using type_for_get = typename type_for_get_helper<
     typename get_matching_tag<GlobalCacheTag, Metavariables>::type>::type;
 
-template <class T, class = std::void_t<>>
-struct has_component_being_mocked_alias : std::false_type {};
-
-template <class T>
-struct has_component_being_mocked_alias<
-    T, std::void_t<typename T::component_being_mocked>> : std::true_type {};
-
-template <class T>
-constexpr bool has_component_being_mocked_alias_v =
-    has_component_being_mocked_alias<T>::value;
-
-template <typename ComponentToFind, typename ComponentFromList>
-struct get_component_if_mocked_helper {
-  static_assert(
-      has_component_being_mocked_alias_v<ComponentFromList>,
-      "The parallel component was not found, and it looks like it is not being "
-      "mocked. Did you forget to add it to the "
-      "'Metavariables::component_list'? See the first template parameter for "
-      "the component that we are looking for and the second template parameter "
-      "for the component that is being checked for mocking it.");
```

Comment:
If I remember correctly I added this assert to help with the common user error where you forget to add a parallel component to `component_list`. I think with the current change you will get a cryptic "can't do tmpl::front on an empty list" error or something like that. Could you please make sure the scenario gives a useful error?